File: core/src/server/model/mapData/GameOption.scala
package server.model.mapData

import engine.general.network.DisplayCommunicator
import server.controller.playerControls.HumanPlayer
import server.model.playerData.Player
import javax.swing._
import java.util.ArrayList
import engine.rts.model.OptionBase

class GameOption(numSlider:JSlider,val nTroop:Boolean) extends OptionBase(numSlider){

    var players=new ArrayList[HumanPlayer]//This represents a list of players that have connected to the client
    def neutralTroop()=nTroop
    def addPlayer(com:DisplayCommunicator){
        players.add(new HumanPlayer(com))
    }
    def getClients()=players
}

File: core/src/server/model/mapData/GameOption.java
package server.model.mapData;

import engine.general.network.DisplayCommunicator;
import server.controller.playerControls.HumanPlayer;
import java.util.ArrayList;
import engine.rts.model.OptionBase;

public class GameOption extends OptionBase{
	
    ArrayList<HumanPlayer> players=new ArrayList<HumanPlayer>();//This represents a list of players that have connected to the client
    boolean neutralTroops;
	public GameOption(int numPlayers,boolean nTroop){
		super(numPlayers);
		neutralTroops=nTroop;
	}

    public boolean neutralTroop(){
    	return neutralTroops;
    }
    
    public void addPlayer(DisplayCommunicator com){
        players.add(new HumanPlayer(com));
    }
    
    public ArrayList<HumanPlayer> getClients(){
    	return players;
    }
}